pkg/daemon/cleanup/cleanup.go

```diff
@@ -92,18 +94,18 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 }
 
 // getMergedCMData attempts to get the latest and the version CM data specified by the node's version annotation
-// It returns the merged CM Data containing services and the watched environment variables
+// It returns the merged CM Data containing services, the watched environment variables, and if best effort is desired
 func getMergedCMData(ctx context.Context, cli client.Client,
```

Comment:
This function seems a bit overloaded with this addition.
A teller of this is that 3 things are being returned.

Reply:
Its difficult to tell the criteria of whether a cleanup should be best effort or not.

Reply:
Can best effort cleanup always be done? Is a bool even required?

Reply:
>  A teller of this is that 3 things are being returned.

Note: not saying this is always the case, but at the least it means we should inspect whats happening here closely to see if its truely required.

Reply:
explored an alternative with [these drafted commits](https://github.com/openshift/windows-machine-config-operator/compare/master...saifshaikh48:windows-machine-config-operator:managed-svc-cleanup2?expand=1). 

resolving the convo as we decided to keep this approach. I've since improved the docs around the logic/decisions